### ISSUE TITLE
[Bug] Open application windows on new messages don't work

### ISSUE THREAD

**Issue by ankar84** [RocketChat/Rocket.Chat.Electron#1909]
## My Setup

- Operating System:
- App Version: 3.0.7
- Installation type: exe

- [x] I have tested with the latest version
- [x] I can simulate the issue easily

## Description
There is option in desktop application - Window => Maximize windows on new messages (I don't know what actually it on English, I have a Russian version `Развернуть при поступлении новых сообщений`) but I see, that now in 3.0.7 it doesn't work.
Our users need that feature.

### Current Behavior
That setting don't work
![image](https://user-images.githubusercontent.com/4023037/101621419-c698c100-3a26-11eb-99ce-d10ba7b1ec53.png)

### Expected Behavior
Users need that Electron application open and maximized when new messages appear in chats.
And very important thing is - it would be perfect if that option will be part of notifications system and could be configured on a per chat basis (like other notifications - mobile, desktop, email - as option of desktop settings)

**Comment by ankar84** [comment]
still an issue in 3.1.1

**Comment by ankar84** [comment]
Any news?

**Comment by ankar84** [comment]
Not fixed in 3.2.2

**Comment by ankar84** [comment]
Not working in 3.7.0 pre release. Blinking but not maximazing.

**Comment by jeanfbrito** [comment]
It works but just don't maximize? We need it to add a maximize behavior then?

**Comment by ankar84** [comment]
> It works but just don't maximize? We need it to add a maximize behavior then?

No, in 3.7.0 Electron window not opens on message. It blinking in System Taskbar, but not open (maximizing).

That issue is only about **Window => Maximize windows on new messages** feature.
![image](https://user-images.githubusercontent.com/4023037/144593252-6f7d3dce-8538-4f2d-bd21-4a9ead20c7cf.png)

**Comment by jeanfbrito** [comment]
Thats fixed right?

**Comment by ankar84** [comment]
> Thats fixed right? 

No, Jean!
Unfortunately that issue is still exists in most recent 3.5.7 version.

**Comment by ankar84** [comment]
I just test it again and can tell that bug fixed partly in 3.5.7
That option works perfect if you Minimize app in system tray (near clock right bottom). When message arrived in chat, that app maximize from tray and get on top of screen.
But if app minimized to task bar (with minimize button) of just appears in a background of other app - that feature don't work.

But I believe it should maximize and become on top in all cases.

![max-01](https://user-images.githubusercontent.com/4023037/149276582-8756c220-0a87-40f2-8ab5-04b0c3f9e9c2.gif)

@jeanfbrito can you check it?

**Comment by jeanfbrito** [comment]
Sorry I forgot to reply. I will take a look ASAP

**Comment by ankar84** [comment]
in 3.7.8 that feature don't work at all again like it was in 3.0.7 (first post)

**Comment by ankar84** [comment]
In 3.7.9 behavior of application is a bit different.
Now both close and minimize buttons are minimizing application to system tray.
When message arrived to app, it start blinking in task bar and notification toast appears, but application windows not maximizing.

**Comment by ankar84** [comment]
In 3.8.0 behavior of that option `Open application windows on new messages` is [same](https://github.com/RocketChat/Rocket.Chat.Electron/issues/1909#issuecomment-1011832706) as in 3.5.7
That option works perfect if you Minimize app in system tray (near clock right bottom) with X button, like close. When message arrived in chat, that app maximize from tray and get on top of screen.
But if app minimized to task bar (with minimize button) of just appears in a background of other app - that feature don't work.

**Comment by ankar84** [comment]
Still an issue in 3.8.8

**Comment by ankar84** [comment]
@jeanfbrito in 3.8.11 that feature still doesn't work as expected.
If option `Open application windows on new messages` is enabled, on a new message app is highlighted in taskbar
![image](https://user-images.githubusercontent.com/4023037/192713883-99c94b43-ba56-4d71-8f30-c62f0a45ea2b.png)
If option `Open application windows on new messages` is disabled, on a new message app is not highlighted in taskbar
![image](https://user-images.githubusercontent.com/4023037/192714137-f4123188-fbde-4ff3-9d44-220419e35fc3.png)

But there is no maximization of window in any case.
